File: dotnet/cafdemalihapa/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/EnvironmentVariablesTemplates/EnvironmentVariablesTemplatesDirectory.cs
namespace cafdemalihapa.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesTemplates
{
    public class EnvironmentVariablesTemplatesDirectory(
        Alone.AloneDirectory aloneDirectory,
        ScriptsDirectory scriptsDirectory
    )
    {
        public string GetPath()
        {
            var templateSourceDirectory_construction =
                Path.Combine(
                    scriptsDirectory.GetPath(),
                    "environment-variables-template-files"
                );

            var templateSourceDirectory = aloneDirectory.IsAlone()
                ? aloneDirectory.GetName()
                : templateSourceDirectory_construction;

            return templateSourceDirectory;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Repository/Dotnet/Scripts/EnvironmentVariablesTemplates/EnvironmentVariablesTemplatesDirectory.fs
namespace cafdemwimu.console.Directories.Repository.Dotnet.Scripts.EnvironmentVariablesTemplates

open System.IO
open cafdemwimu.console.Directories.Repository.Dotnet.Scripts
open cafdemwimu.console.Directories.Repository.Dotnet.Scripts.Alone

type EnvironmentVariablesTemplatesDirectory(aloneDirectory: AloneDirectory, scriptsDirectory: ScriptsDirectory) =
    member _.GetPath() =
        let templateSourceDirectory_construction =
            Path.Combine(scriptsDirectory.GetPath(), "environment-variables-template-files")

        let templateSourceDirectory =
            if aloneDirectory.IsAlone() then aloneDirectory.GetName()
            else templateSourceDirectory_construction

        templateSourceDirectory
